plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/Completion/Completer/Paths.hs

```diff
@@ -0,0 +1,107 @@
+module Ide.Plugin.Cabal.Completion.Completer.Paths where
+
+import qualified Data.List                         as List
+import qualified Data.Text                         as T
+import           Distribution.PackageDescription   (Benchmark (..),
+                                                    BuildInfo (..),
+                                                    CondTree (condTreeData),
+                                                    Executable (..),
+                                                    GenericPackageDescription (..),
+                                                    Library (..),
+                                                    UnqualComponentName,
+                                                    mkUnqualComponentName,
+                                                    testBuildInfo)
+import           Distribution.Utils.Path           (getSymbolicPath)
+import           Ide.Plugin.Cabal.Completion.Types
+import qualified System.FilePath                   as FP
+import qualified System.FilePath.Posix             as Posix
+
+
+-- | Information used to query and build path completions.
+--
+--  Note that pathSegment  combined with queryDirectory  results in
+--  the original prefix.
+--
+--  Example:
+--  When given the written prefix, @dir1\/dir2\/fi@, the
+--  resulting PathCompletionInfo would be:
+--
+--  @
+--    pathSegment = "fi"
+--    queryDirectory  = "dir1\/dir2\/fi"
+--    ...
+--  @
+data PathCompletionInfo = PathCompletionInfo
+  { -- | partly written segment of the next part of the path
+    pathSegment          :: T.Text,
+    -- | written part of path, platform dependent
+    queryDirectory       :: FilePath,
```

Comment:
I don't think the query directory is platform-dependent, at least the tests assume Posix format.